### PR TITLE
Fix typo in README.md regarding Flatcar installer

### LINKED ISSUE #123: 🔍 test: add coverage tests for 7 uncovered functions (batch 2)
## 🔍 Coverage gaps — batch 2 (ready to merge)

**Branch**: `test/coverage-gaps-batch2` (commit `0b85d6d`) — _push blocked, see below_
**Coverage before**: 83.8% total | **Target**: 91%

### What this fixes

7 zero-coverage functions identified by `go tool cover -func=cover.out | grep 0.0%`:

| Package | Function | Coverage before | Est. after |
|---------|----------|----------------|-----------|
| validate | `Gateway()`, `DNSServer()` | 0% | ~97% pkg |
| bakery | `NewHTTPClient()` | 0% | ~85.5% pkg |
| bakery | `FetchChannelInfo()`, `FetchAllChannels()` | 0% | ~86% pkg |
| probe | `DetectNvidiaGPUs()` smoke | 0% | ~86% pkg |
| tui | `viewInstall()` (3 branches) | 0% | ~80% pkg |
| tui | `sysextItem.FilterValue()`, `sysextDelegate.Update()` | 0% | ~80% pkg |
| wizard | `FetchChannels()` | 0% | ~91% pkg ✅ |

### Files changed

```
internal/validate/validate_test.go    +42 lines
internal/bakery/bakery_test.go        +13 lines
internal/bakery/channels_test.go      +24 lines
internal/probe/probe_test.go          +8 lines
internal/tui/tui_coverage_test.go     +86 lines (new file)
internal/wizard/wizard_test.go        +13 lines
```

### To apply

```bash
# Option A — apply patch directly
curl -sL https://github.com/projectbluefin/knuckle/issues/ISSUE_NUM | ... # see patch below
git am < coverage-gaps-batch2.patch

# Option B — cherry-pick from local agent worktree
# /data/agents/quality/knuckle-pr2  commit 0b85d6d
```

### Push blocker

The `kubestellar-hive` GitHub App installation on this repo lacks `contents: write` permission. Granting it will enable future automated PRs.

<details>
<summary>📎 Full patch (apply with <code>git am</code>)</summary>

```patch
From 0b85d6d2cf5e8873f9807731028702246a5f3c87 Mon Sep 17 00:00:00 2001
From: "kubestellar-hive[bot]" <kubestellar-hive[bot]@users.noreply.github.com>
Date: Fri, 22 May 2026 08:49:55 -0400
Subject: [PATCH] test: add coverage tests for 7 uncovered functions
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Findings from measured coverage run (83.8% total → target 91%):

- validate: Gateway() and DNSServer() — both delegate to IPAddress()
  but had zero direct test coverage; add table-driven tests for valid
  and invalid inputs.

- bakery: NewHTTPClient() — production constructor always skipped in
  favour of NewHTTPClientWithURL(); add a basic smoke test.

- bakery: FetchChannelInfo() and FetchAllChannels() — public amd64-
  defaulting wrappers tested only via internal helpers; add cancelled-
  context delegation tests that exercise the wrapper without network.

- probe: DetectNvidiaGPUs() — thin wrapper around nvidiaGPUsFromClient
  with a real nvpci client; add a no-panic smoke test (nil is valid on
  CI where no GPU is present).

- tui: viewInstall() — three-branch rendering function (prompt /
  progress messages / installing spinner) with zero prior coverage;
  add one test per branch.

- tui: sysextItem.FilterValue() and sysextDelegate.Update() — both in
  sysext_list.go which lacked any direct tests for these two methods;
  add targeted tests in new tui_coverage_test.go.

- wizard: FetchChannels() — calls bakery.FetchAllChannels() with no
  mock path; test error propagation via pre-cancelled context.

Signed-off-by: kubestellar-hive[bot] <kubestellar-hive[bot]@users.noreply.github.com>
---
 internal/bakery/bakery_test.go     | 13 +++++
 internal/bakery/channels_test.go   | 24 +++++++++
 internal/probe/probe_test.go       |  8 +++
 internal/tui/tui_coverage_test.go  | 86 ++++++++++++++++++++++++++++++
 internal/validate/validate_test.go | 42 +++++++++++++++
 internal/wizard/wizard_test.go     | 13 +++++
 6 files changed, 186 insertions(+)
 create mode 100644 internal/tui/tui_coverage_test.go

diff --git a/internal/bakery/bakery_test.go b/internal/bakery/bakery_test.go
index 8b49a90..d6adcd1 100644
--- a/internal/bakery/bakery_test.go
+++ b/internal/bakery/bakery_test.go
@@ -561,3 +561,16 @@ func TestFetchSHA256MissingAsset(t *testing.T) {
 		t.Errorf("expected empty Sha256 when no SHA256SUMS asset, got %q", entries[0].Sha256)
 	}
 }
+
+func TestNewHTTPClient(t *testing.T) {
+c := bakery.NewHTTPClient()
+if c == nil {
+t.Fatal("NewHTTPClient() returned nil")
+}
+if c.CatalogURL != bakery.DefaultCatalogURL {
+t.Errorf("CatalogURL = %q, want %q", c.CatalogURL, bakery.DefaultCatalogURL)
+}
+if c.HTTP == nil {
+t.Error("HTTP client field is nil")
+}
+}
diff --git a/internal/bakery/channels_test.go b/internal/bakery/channels_test.go
index 7a5a6bf..453ef14 100644
--- a/internal/bakery/channels_test.go
+++ b/internal/bakery/channels_test.go
@@ -361,3 +361,27 @@ func TestFetchAllChannelsArch_InvalidArch(t *testing.T) {
 		t.Fatal("expected error for unsupported arch")
 	}
 }
+
+// TestFetchChannelInfoWrapper exercises the FetchChannelInfo public wrapper
+// (amd64-defaulting) which delegates to FetchChannelInfoArch. Uses a cancelled
+// context to trigger an error without making a real network call.
+func TestFetchChannelInfoWrapper(t *testing.T) {
+ctx, cancel := context.WithCancel(context.Background())
+cancel()
+_, err := FetchChannelInfo(ctx, "stable")
+if err == nil {
+t.Fatal("FetchChannelInfo with cancelled context: expected error, got nil")
+}
+}
+
+// TestFetchAllChannelsWrapper exercises the FetchAllChannels public wrapper
+// (amd64-defaulting) which delegates to FetchAllChannelsArch. Uses a cancelled
+// context to trigger an error without making a real network call.
+func TestFetchAllChannelsWrapper(t *testing.T) {
+ctx, cancel := context.WithCancel(context.Background())
+cancel()
+_, err := FetchAllChannels(ctx)
+if err == nil {
+t.Fatal("FetchAllChannels with cancelled context: expected error, got nil")
+}
+}
diff --git a/internal/probe/probe_test.go b/internal/probe/probe_test.go
index 266a672..3b47b50 100644
--- a/internal/probe/probe_test.go
+++ b/internal/probe/probe_test.go
@@ -286,3 +286,11 @@ func TestDetectNvidiaGPUs_ErrorFromClient(t *testing.T) {
 		t.Errorf("expected nil on error, got %v", gpus)
 	}
 }
+
+// TestDetectNvidiaGPUs_Smoke calls the public wrapper around nvidiaGPUsFromClient.
+// On CI (no NVIDIA hardware) it returns nil — that is a valid result.
+// This test only verifies it does not panic.
+func TestDetectNvidiaGPUs_Smoke(t *testing.T) {
+result := DetectNvidiaGPUs()
+_ = result // nil is valid when no GPU is present
+}
diff --git a/internal/tui/tui_coverage_test.go b/internal/tui/tui_coverage_test.go
new file mode 100644
index 0000000..9031005
--- /dev/null
+++ b/internal/tui/tui_coverage_test.go
@@ -0,0 +1,86 @@
+package tui
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/projectbluefin/knuckle/internal/bakery"
+	"github.com/projectbluefin/knuckle/internal/model"
+)
+
+// ── Finding 5: viewInstall branches ──────────────────────────────────────────
+
+// TestViewInstallPrompt covers the "not installing, no progress" branch that
+// prompts the user to press Enter.
+func TestViewInstallPrompt(t *testing.T) {
+	m := New(newTestWizard())
+	m.installing = false
+	m.Wizard.State.ProgressMessages = nil
+
+	out := m.viewInstall()
+	if !strings.Contains(out, "Press Enter to start installation") {
+		t.Errorf("initial state: %q missing press-enter prompt", out)
+	}
+}
+
+// TestViewInstallWithProgress covers the progress-messages branch (completed
+// phases rendered with checkmarks).
+func TestViewInstallWithProgress(t *testing.T) {
+	m := New(newTestWizard())
+	m.installing = false
+	m.Wizard.State.ProgressMessages = []string{
+		"Writing ignition config",
+		"Running flatcar-install",
+	}
+
+	out := m.viewInstall()
+	if !strings.Contains(out, "Writing ignition config") {
+		t.Errorf("progress not rendered: %q", out)
+	}
+	if !strings.Contains(out, "Running flatcar-install") {
+		t.Errorf("second progress message missing: %q", out)
+	}
+}
+
+// TestViewInstallWhileInstalling covers the m.installing == true branch
+// (spinner + progress bar rendered).
+func TestViewInstallWhileInstalling(t *testing.T) {
+	m := New(newTestWizard())
+	m.installing = true
+
+	out := m.viewInstall()
+	// The spinner text ("Working...") must appear while installing.
+	if !strings.Contains(out, "Working") {
+		t.Errorf("installing state missing spinner/working text: %q", out)
+	}
+}
+
+// ── Finding 6: sysext_list FilterValue and delegate.Update ───────────────────
+
+// TestSysextItemFilterValue verifies FilterValue concatenates Name, Category,
+// and SupportTier so the bubbles/list filter can match on all three fields.
+func TestSysextItemFilterValue(t *testing.T) {
+	item := sysextItem{
+		idx: 0,
+		entry: model.SysextEntry{
+			Name:        "nvidia",
+			Category:    "GPU",
+			SupportTier: bakery.TierIntegrated,
+		},
+	}
+	fv := item.FilterValue()
+	for _, want := range []string{"nvidia", "GPU", bakery.TierIntegrated} {
+		if !strings.Contains(fv, want) {
+			t.Errorf("FilterValue() = %q, missing %q", fv, want)
+		}
+	}
+}
+
+// TestSysextDelegateUpdate verifies the no-op Update implementation returns nil.
+func TestSysextDelegateUpdate(t *testing.T) {
+	d := newSysextDelegate(func(int) bool { return false })
+	cmd := d.Update(nil, nil)
+	if cmd != nil {
+		t.Errorf("Update() = %v, want nil (no-op)", cmd)
+	}
+}
diff --git a/internal/validate/validate_test.go b/internal/validate/validate_test.go
index b61597a..634e6db 100644
--- a/internal/validate/validate_test.go
+++ b/internal/validate/validate_test.go
@@ -519,3 +519,45 @@ func TestCheckConsistency(t *testing.T) {
 		})
 	}
 }
+
+func TestGateway(t *testing.T) {
+tests := []struct {
+input   string
+wantErr bool
+}{
+{"192.168.1.1", false},
+{"10.0.0.1", false},
+{"not-an-ip", true},
+{"", true},
+{"256.0.0.1", true},
+}
+for _, tt := range tests {
+t.Run(tt.input, func(t *testing.T) {
+err := Gateway(tt.input)
+if (err != nil) != tt.wantErr {
+t.Errorf("Gateway(%q) error = %v, wantErr %v", tt.input, err, tt.wantErr)
+}
+})
+}
+}
+
+func TestDNSServer(t *testing.T) {
+tests := []struct {
+input   string
+wantErr bool
+}{
+{"8.8.8.8", false},
+{"1.1.1.1", false},
+{"not-dns", true},
+{"", true},
+{"300.0.0.1", true},
+}
+for _, tt := range tests {
+t.Run(tt.input, func(t *testing.T) {
+err := DNSServer(tt.input)
+if (err != nil) != tt.wantErr {
+t.Errorf("DNSServer(%q) error = %v, wantErr %v", tt.input, err, tt.wantErr)
+}
+})
+}
+}
diff --git a/internal/wizard/wizard_test.go b/internal/wizard/wizard_test.go
index a2f6147..774abe6 100644
--- a/internal/wizard/wizard_test.go
+++ b/internal/wizard/wizard_test.go
@@ -1864,3 +1864,16 @@ func TestRouteE_ReviewBackToUser_FixAndReadvance(t *testing.T) {
 		t.Error("Butane does not contain the fixed SSH key")
 	}
 }
+
+// TestFetchChannelsCancelledContext verifies that FetchChannels propagates the
+// error returned by bakery.FetchAllChannels. A pre-cancelled context triggers
+// an immediate network error without making a real HTTP call.
+func TestFetchChannelsCancelledContext(t *testing.T) {
+w, _, _, _ := newTestWizard()
+ctx, cancel := context.WithCancel(context.Background())
+cancel()
+err := w.FetchChannels(ctx)
+if err == nil {
+t.Fatal("FetchChannels with cancelled context: expected error, got nil")
+}
+}
-- 
2.39.5
```
</details>

### PR DESCRIPTION
**Pull request by nikodunk**
## What
Fixes a typo in the readme

## How tested

- [ ] `just ci` is green locally
- [ ] `just vm-e2e` passes (end-to-end install + boot)
- [ ] `just headless-test` passes (for headless-mode changes)
- [ ] Tested on real hardware — describe below
- [x] Doc / template / CI change only — no install path touched

## Notes for reviewers

---

## Summary by CodeRabbit

* **Documentation**
  * Fixed a typo in the README's Knuckle description.

[Review Change Stack](https://app.coderabbit.ai/change-stack/projectbluefin/knuckle/pull/255?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by castrojo — APPROVED** [review]
Typo fix verified: "Flatcat" → "Flatcar" in README.md. Correct.

Assisted-by: claude-sonnet-4-5 via pi

**Review by hanthor — APPROVED** [review]

**Review by castrojo — APPROVED** [review]
✅ Typo fix verified in diff: 'Flatcat' → 'Flatcar'. LGTM.
